### ISSUE TITLE
fix: 🤔 implement `sd-select` support for TalkBack

### ISSUE THREAD

**Issue by paulovareiro29** [solid-design-system/solid#1708]
## Current behavior
When using TalkBack on mobile Android devices, the `sd-select` is not accessible. The user is unable to open the options list.

## Expected behavior
The `sd-select` is fully accessible using TalkBack on Android devices.

## Steps to reproduce
- With an android device, turn on TalkBack.
- Go to the [`sd-select story`](https://solid-design-system.github.io/solid/main/iframe.html?args=&globals=&id=components-sd-select-screenshots-sd-select--default&viewMode=story)
- Try to interact with the `sd-select`

### Suggested Reproduction Tools

- **sd-select story**: https://solid-design-system.github.io/solid/main/iframe.html?args=&globals=&id=components-sd-select-screenshots-sd-select--default&viewMode=story

## Technical Information
From testing multiple design systems, only the [Listbox from Headless UI](https://headlessui.com/v1/vue/listbox) is fully accessible using TalkBack.
I have noticed they use a `<button />` as a trigger for the popup. We are currently using div / input.
We might need to restructure our `sd-select` component to use a `<button />` as a trigger instead, and hide the input just to keep support when using the component within forms

## DoR
- [x] Item has business value
- [x] Item has been estimated by the team
- [x] Item is clear and well-defined
- [x] Item dependencies have been identified

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch
